lang/java/reef-io/src/test/java/org/apache/reef/io/network/NetworkServiceTest.java

```diff
@@ -546,17 +560,13 @@ public void testMessagingNetworkServiceBatchingRate() throws Exception {
     @Override
     public void onNext(final Message<T> value) {
 
-      count.incrementAndGet();
+      final int currentCount = count.incrementAndGet();
 
-      LOG.log(Level.FINEST,
-          "OUT: {0} received {1} from {2} to {3}",
-          new Object[]{name, value.getData(), value.getSrcId(), value.getDestId()});
+      LOG.log(Level.FINER, "{0} Message {1}/{2} :: {3}", new Object[] {name, currentCount, expected, value});
 
-      for (final T obj : value.getData()) {
-        LOG.log(Level.FINEST, "OUT: data: {0}", obj);
-      }
+      Assert.assertTrue(currentCount <= expected);
 
-      if (count.get() == expected) {
+      if (currentCount >= expected) {
```

Comment:
Why do you use `>=` instead of `==`? I think `currentCount` cannot be greater than `expected` because of the above assertion.